src/monitor-plan/monitor-plan.service.ts

```diff
@@ -128,14 +130,12 @@ export class MonitorPlanService {
 
     if (getReportingFrquencies) {
       REPORTING_FREQ = 0;
-      promises.push(
-        this.reportingFreqRepository.findBy({ monitorPlanId: planId }),
-      );
+      promises.push(  await useSlaveRepository(this.dataSource, MonitorPlanReportingFrequencyRepository, async (repository) => repository.findBy({ monitorPlanId: planId })));
```

Comment:
We don't want to await this here, because `Promise.all` is used later so they are executed in parallel.
```suggestion
      promises.push(useSlaveRepository(this.dataSource, MonitorPlanReportingFrequencyRepository, async (repository) => repository.findBy({ monitorPlanId: planId })));
```

Reply:
updated this in latest commit